src/refactor_nrepl/ns/slam/hound/search.clj

```diff
@@ -121,18 +121,18 @@
         (all-classpath-entries)))
 
 (def available-classes
-  (get-available-classes))
+  (delay (get-available-classes)))
 
 (defn- get-available-classes-by-last-segment
   []
   (delay
-   (group-by #(symbol (peek (string/split (str %) #"\."))) available-classes)))
+   (group-by #(symbol (peek (string/split (str %) #"\."))) @available-classes)))
 
 (def available-classes-by-last-segment
-  (get-available-classes-by-last-segment))
+  (delay (get-available-classes-by-last-segment)))
 
 (defn reset
   "Reset the cache of classes"
   []
-  (alter-var-root #'available-classes (constantly (get-available-classes)))
-  (alter-var-root #'available-classes-by-last-segment (constantly (get-available-classes-by-last-segment))))
+  (alter-var-root #'available-classes (constantly (delay (get-available-classes))))
```

Comment:
Unless I'm reading it wrong you'll get a delay in a delay here (and on the line below)?